### PR TITLE
added opt-out for autotune

### PR DESCRIPTION
**Pull request by koenhelwegen**
Autotune is still experimental so I think it would be good to provide the possibility to opt-out (I personally would like this for testing purposes).

### CODE REVIEW

File: zookeeper/data.py [hunk 1]
```diff
@@ -114,7 +117,12 @@ def test_data(self, batch_size):
...
-            .map(self.map_fn, num_parallel_calls=tf.data.experimental.AUTOTUNE)
+            .map(
+                self.map_fn,
+                num_parallel_calls=tf.data.experimental.AUTOTUNE
+                if self.autotune
+                else 1 * batch_size,
+            )
             .batch(batch_size)
-            .prefetch(tf.data.experimental.AUTOTUNE)
+            .prefetch(tf.data.experimental.AUTOTUNE if self.autotune else 2)
```
Comment (jamescook106) [inline zookeeper/data.py]: Would we be able to set the prefetch amount to be configurable? On a side note I think before it was around 8

File: zookeeper/cli.py [hunk 2]
```diff
@@ -61,6 +61,11 @@ def build_train(function):
         default=False,
         help="If you want to split a dataset which only contains a train/test into train/val/test",
     )
+    @click.option(
+        "--autotune/--no-autotune",
+        default=True,
+        help="Use tf.data.experimental.AUTOTUNE for data pipeline.",
+    )
```
Comment (jamescook106) [inline zookeeper/cli.py]: Please add a usage example to the docs explaining why one may not want to autotune

File: zookeeper/data.py [hunk 3]
```diff
@@ -97,10 +98,12 @@ def train_data(self, batch_size):
             .repeat()
             .map(
                 functools.partial(self.map_fn, training=True),
-                num_parallel_calls=tf.data.experimental.AUTOTUNE,
+                num_parallel_calls=tf.data.experimental.AUTOTUNE
+                if self.autotune
+                else 1 * batch_size,
             )
```
Comment (jamescook106) [inline zookeeper/data.py]: is 1* batch_size the correct ration?

Comment (koenhelwegen) [reply zookeeper/data.py]: TF recommends using the number of CPU cores, but in my experience using order batch sizes gives slightly better results.

File: zookeeper/data.py (hunk 1 above)
Comment (koenhelwegen) [reply zookeeper/data.py]: @lgeiger how do you feel about this?

**Comment by lgeiger** [comment]
I am a bit unsure, about this. I think we should definitely give people the opportunity to customize this, but at least for autotuning TensorFlow datasets actually uses autotuning under the hood as well so this flag might be a bit confusing. I'll take a closer look at this tomorrow.

**Comment by lgeiger** [comment]
Closing this for now, until we can come up with a general solution.

**Pull request closed without merging**